### PR TITLE
add docstring for `DirichletMultinomial`

### LINKED ISSUE #1601: `DirichletMultinomial` missing docstring
There's no documentation for `DirichletMultinomial`, and it's also not clear to me what `α0` is supposed to be--the usual parametrization of a `DirichletMultinomial` is just in terms of `n` and `α`.

### PR DESCRIPTION
**Pull request by pri1311**
Closes #1601 

Latex output image from QuickLatex render:
Screenshot 2022-12-30 at 10 16 22 PM

### CODE REVIEW

File: src/multivariate/dirichletmultinomial.jl [hunk 1]
```diff
@@ -1,3 +1,24 @@
+"""
+The [Dirichlet-Multinomial distribution](https://en.wikipedia.org/wiki/Dirichlet-multinomial_distribution)
+can be understood as draws from a Multinomial distribution where each sample has a slightly different probability vector, which is itself drawn from a common Dirichlet distribution. This contrasts with the Multinomial distribution, which assumes that all observations arise from a single fixed probability vector. This enables the Dirichlet-multinomial to accommodate more variable (a.k.a, over-dispersed) count data than the Multinomial.
```
Comment (ParadaCarleton) [inline src/multivariate/dirichletmultinomial.jl:3]:
```suggestion
is the distribution of a draw from a Multinomial distribution where each sample has a slightly different probability vector,  drawn from a common Dirichlet distribution. 

This contrasts with the Multinomial distribution, which assumes that all observations arise from a single fixed probability vector. This enables the Dirichlet-multinomial to accommodate more variable (a.k.a, over-dispersed) count data than the Multinomial.
```

File: src/multivariate/dirichletmultinomial.jl [hunk 2]
```diff
@@ -1,3 +1,24 @@
+"""
```
Comment (ParadaCarleton) [inline src/multivariate/dirichletmultinomial.jl:1]: Docstring is missing a type signature--see [here for examples.](https://github.com/invenia/BlueStyle#documentation)

**Review by ParadaCarleton — CHANGES_REQUESTED** [review]

File: src/multivariate/dirichletmultinomial.jl [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+where
+- ``x_k`` is the k-th count
+- ``n = \\sum_k x_k``
+- ``\\alpha_k`` is a prior belief about the k-th count
+- ``\\alpha_0 = \\sum_k \\alpha_k``
+- ``\\Gamma`` is the gamma function, defined as ``\\Gamma(x) = (x -1)!``
+
+```julia
+DirichletMultinomial(n::Integer, α::Vector{T}) where {T <: Real}   # Dirichlet-Multinomial distribution for n trials with parameter vector α.
+DirichletMultinomial(n::Integer, k::Integer)                       # Dirichlet-Multinomial distribution with n trials and paramemeter vector of length k of ones.
```
Comment (devmotion) [inline src/multivariate/dirichletmultinomial.jl:21]:
```suggestion

# Let k be a positive integer
DirichletMultinomial(n, k) # Dirichlet-multinomial distribution with n trials and parameter vector of length k of ones.
```

**Review by ParadaCarleton — APPROVED** [review]
Looks good to me! @devmotion any final comments?

**Pull request merged**